nixos/modules/services/x11/window-managers/qtile.nix

```diff
@@ -7,19 +7,34 @@ let
 in
 
 {
-  options = {
-    services.xserver.windowManager.qtile.enable = mkEnableOption "qtile";
+  options.services.xserver.windowManager.qtile = {
+    enable = mkEnableOption "qtile";
+
+    package = mkOption {
+      type = package;
+      default = pkgs.qtile;
+      defaultText = "pkgs.qtile";
+      description = ''
+          The qtile package to be ran.
+      '';
+    };
+
   };
 
   config = mkIf cfg.enable {
     services.xserver.windowManager.session = [{
       name = "qtile";
       start = ''
-        ${pkgs.qtile}/bin/qtile start &
+        ${cfg.package}/bin/qtile start &
         waitPID=$!
       '';
     }];
 
-    environment.systemPackages = [ pkgs.qtile ];
+    environment.systemPackages = [
+      # pkgs.qtile is currently a buildenv of qtile and its dependencies.
+      # For userland commands, we want the underlying package so that
+      # packages such as python don't bleed into userland and overwrite intended behavior.
+      (pkgs.qtile.unwrapped or pkgs.qtile)
```

Comment:
shouldn't this use `cfg.package.unwrapped` as its first choice? (no idea, not using qtile)

Reply:
yea, it should